### ISSUE TITLE
Suppress HTTPS filtering error notification for the apps that use pinning techniques

### ISSUE THREAD

**Issue by TheHasagi** [AdguardTeam/AdguardForAndroid#3052]
Somehow from time to time user will receive a notification (see attachment)  about HTTPS filtering error that some application does not accept our certificate. It can not be reproduced easily but it does exist.

<details>

![image](https://user-images.githubusercontent.com/37797229/64361509-5cb88d00-d015-11e9-869b-f5bab4ed85d8.png)

</details>

**Comment by ameshkov** [comment]
Does it have HTTPS filtering enabled for Outlook? We don't filter it by default

**Comment by TheHasagi** [comment]
I guess not, can be reproduced with default settings (after a clean install of the application).

**Comment by Eugene-Savenko** [comment]
![image](https://user-images.githubusercontent.com/15856304/64509116-b7890780-d2e7-11e9-92a9-aa8f56a85f05.png)

If this can help, please find the logs here: 2307731

**Comment by TheHasagi** [comment]
New logs in ncloud.

**Comment by 3xploiton3** [comment]
I move cert adguard to system,

￼

**Comment by Murgeye** [comment]
For me it really is not "from time to time" but constantly and it completely breaks HTTP Filtering.

**Comment by Chinaski1** [comment]
@Murgeye 

I suppose this is your mail to support - *2354792*
Thanks for the provided logs.
**Added to NCloud**

**Comment by Murgeye** [comment]
> @Murgeye
> 
> I suppose this is your mail to support - _2354792_
> Thanks for the provided logs.
> **Added to NCloud**

Yes it is, sorry, I forgot to mention it here after sending in a support request.

Btw., there seems to be no good way to work around this bug(?) in Beta except by adding every single App or Domain to the whitelist. Disabling the notifications in the low level settings causes some Apps to be unable to open any TLS connections at all.

**Comment by ameshkov** [comment]
Some of the apps do TLS pinning and HTTPS filtering won't work for them.

Ideally, you should disable HTTPS filtering for such apps.

**Comment by ameshkov** [comment]
@AdguardTeam/developers I think we should have a list of apps known for using TLS pinning and suppress this notification for them.

We should expose this list via low-level settings to make it transparent to the users.

**Comment by Murgeye** [comment]
What was the default behavior before the notification was added? I haven't seen this notification before the last beta update, but had no problems with the same applications. I find it highly unlikely that almost all apps have suddenly started using certificate pinning for domains like crashlytics.com.

Additionally, I was able to use all of them without in-app ads before.

**Comment by Murgeye** [comment]
BTW., the exclude domain button from the notification does not seem to work for me, as it always overrides the last domain added to the HTTPS whitelist. (Therefore, you can only always add a single domain to the whitelist and it is getting replaced by the next one.)

**Comment by AlfredSpijker** [comment]
I can confirm the issue Murgeye mentioned above. Running latest beta 3.3.2.
Is it better to report this separately some else?

**Comment by Murgeye** [comment]
The bug was present in 3.3 beta 1 as well by the way.

**Comment by artemiv4nov** [comment]
This bug will be fixed in beta 3.

**Comment by ameshkov** [comment]
So here's what we have here:

1. We should add a new low-level setting `pref.https.ignore.filtering.error` with the list of apps that are known to use pinning techniques (@artemiv4nov plz post the exact list)
2. We should fix the "exclude" buttons: https://github.com/AdguardTeam/AdguardForAndroid/issues/3052#issuecomment-546257588

**Comment by artemiv4nov** [comment]
Well, there is third corelibs-related trouble.
More about it here: https://github.com/AdguardTeam/CoreLibs/issues/1175

**Comment by trevorbrachtenbach007-dot** [comment]
https://github.com/AdguardTeam/AdguardForAndroid/issues/3052
